Makefile

```diff
@@ -1,4 +1,17 @@
 # ---------------------------------------------------------------------
+# Build configuration
+# ---------------------------------------------------------------------
+
+BUILD_DIRECTORY ?= release
+
+ifndef BUILD_DIRECTORY
```

Comment:
Yeah, that sounds like a good solution. And obviously if `BUILD_DIRECTORY` ever gets mentioned in the documentation, we'd need to mention that there too. But for now just a comment in the Makefile above the BUILD_DIRECTORY declaration is probably enough.